applications/luci-app-softether/htdocs/luci-static/resources/view/softether-status.js

```diff
@@ -0,0 +1,131 @@
+'use strict';
+'require view';
+'require fs';
+
+function setElementShown(selector, shown) {
+	var loaderE = document.querySelector(selector);
+	if (loaderE) {
+		if (loaderE.classList.contains('hidden') === shown)
+		{
+			if (shown) {
+				loaderE.classList.remove('hidden');
+			} else {
+				loaderE.classList.add('hidden');
+			}
+		}
+	}
+}
+function fixupResponse(response) {
+	return response.toString().replace(/(\n$)/, '');
+}
+
+function rpcDownloadAccountLists()
+{
+	return new Promise(function (resolve, fail) {
+		fs.exec_direct('/usr/libexec/vpncmd-call', [ 'account-list' ])
+			.then(function(resp) {
+				var accounts = [];
+				var responseList = fixupResponse(resp).split('\n');
+				responseList.forEach(function(d) {
+					var s = d.split(',');
+					var tmp = {};
+					tmp.name = s[0];
+					tmp.properties = {};
+					tmp.properties['Status'] = s[1];
+					tmp.properties['Remote'] = s[2];
+					tmp.properties['Hub'] = s[3];
+					accounts.push(tmp);
+				})
+				resolve(accounts);
+			});
+	});
+}
+function rpcDownloadAccountStatus(account) {
+	return new Promise(function (resolve, fail) {
+		fs.exec_direct('/usr/libexec/vpncmd-call', [ 'account-status-get', account.name ])
+			.then(function(resp) {
+				var detailList = fixupResponse(resp).split('\n');
+				detailList.forEach(function(d) {
+					var s = d.split(',');
+					if (s.length === 2)
+						account.properties[s[0]] = s[1];
+				})
+				resolve(account);
+			});
+	});
+}
+
+function downloadAllStatus(accountList) {
+	var promises = [];
+	accountList.forEach( function(account) { promises.push(rpcDownloadAccountStatus(account)); });
+	return Promise.all(promises);
+}
+
+function downloadAllAccounts() {
+	return new Promise(function(resolve) {
+		rpcDownloadAccountLists().then( function(accountList) {
+			downloadAllStatus(accountList).then( function(accountListWDetail) { 
+				resolve(accountListWDetail); 
+			});
+		});
+	});
+}
+
+function updateAccountTable(listData)
+{
+	var tableSelector = '#accountTable';
+	var table = isElem(tableSelector) ? tableSelector : document.querySelector(tableSelector);
+	if (listData.length > 0 ) {
+		listData.forEach(function(account) { 
+			table.appendChild(renderAccountRow(account)); 
+		});
+	} else {
+		setElementShown('#emptyLabel', true);
+	}
+	setElementShown('#loader', false);
+}
+
+function renderAccountRow(account) {
+	var properties = [];
+	for(var key in account.properties) {
+		if (account.properties.hasOwnProperty(key)) {
+			properties.push(E('strong', {}, [key + ':']));
+			properties.push(account.properties[key]);
+			properties.push(E('br', {}, []));
+		}
+	}
+	var row = E('div', {'class':'tr cbi-section-table-row'}, [
+				E('div', {'class':'td', 'style': 'width: 20%;vertical-align:top;'}, [
+					E('strong', {}, ['Account:']),
+					account.name
+				]),
+				E('div', {'class':'td'}, properties)
+			]);
+	return row;
+}
+
+return view.extend({
```

Comment:
We would also put all the functions `enderAccountRow`, `updateAccountTable` into the new object like in:
* https://github.com/openwrt/luci/blob/master/modules/luci-mod-status/htdocs/luci-static/resources/view/status/processes.js 
* https://github.com/openwrt/luci/blob/master/modules/luci-mod-status/htdocs/luci-static/resources/view/status/iptables.js

Reply:
Done.